### PR TITLE
SpringDataAndService

### PR DESCRIPTION
**Pull request by Yanetta**

### CODE REVIEW

File: src/main/java/workSpringData/Main.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import workSpringData.repository.CustomersRepository;
+import workSpringData.service.CustomersService;
+import workSpringData.service.CustomersServiceImpl;
+
+public class Main {
+    public static void main(String[] args) {
+        ConfigurableApplicationContext context = new AnnotationConfigApplicationContext("workSpringData");
+
+        CustomersRepository customersRepository = context.getBean(CustomersRepository.class);
+        customersRepository.findAll().forEach(System.out::println);
```
Comment (DadizhaO) [inline src/main/java/workSpringData/Main.java:14]: You should use CustomersService, not CustomersRepository.

**Review by DadizhaO — APPROVED** [review]

File: src/main/java/entities/Salesreps.java [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+                "empl_num=" + empl_num +
+                "," + "\n" + "name='" + name + '\'' +
+                "," + "\n" + "title='" + title + '\'' +
+                //"," + "\n" + "offices=" + offices +
+                ", " + "\n" + "hire_date=" + hire_date +
+                "," + "\n" + "age=" + age +
+                ", " + "\n" + "sales=" + sales +
+                ", " + "\n" + "quota=" + quota +
+                //", salesreps=" + salesreps +
+//                ","+"\n"+"officesSet=" + officesSet +
```
Comment (SerhiiPasko) [inline src/main/java/entities/Salesreps.java:190]: Clean

File: src/main/java/workSpringData/Main.java (hunk 1 above)
Comment (SerhiiPasko) [reply src/main/java/workSpringData/Main.java:14]: It depends to you.

File: src/main/java/workSpringData/service/CustomersServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import org.springframework.transaction.annotation.Transactional;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import workSpringData.exeption.DeleteException;
+import workSpringData.repository.CustomersRepository;
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.Set;
+
+@Service("service")
```
Comment (SerhiiPasko) [inline src/main/java/workSpringData/service/CustomersServiceImpl.java]: Bad name

Comment (SerhiiPasko) [reply src/main/java/workSpringData/service/CustomersServiceImpl.java]: In our case it doesn't need

File: src/main/java/workSpringData/Main.java (hunk 1 above)
Comment (Yanetta) [reply src/main/java/workSpringData/Main.java:14]: yes, I use it below. Repository was used just to try that it works))

File: src/main/resources/log4j2.xml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    </Properties>
+    <Appenders>
+        <Console name="STDOUT" target="SYSTEM_OUT" follow="true">
+            <PatternLayout pattern="${LOG_PATTERN}"/>
+        </Console>
+    </Appenders>
+    <Loggers>
+
+        <AsyncLogger sort="error-logger" level="ERROR">
+            <AppenderRef ref="FEMAIL"/>
```
Comment (SerhiiPasko) [inline src/main/resources/log4j2.xml]: Where your appender "Femail"?

**Review by SerhiiPasko — COMMENTED** [review]
Not any critical - but read comments

**Pull request merged**